macro/macroutils.py

```diff
@@ -247,13 +252,51 @@ def load_repo_devel_job_data(repo_name, distro=None):
     """
     return _load_manifest_file(repo_devel_job_data_file(repo_name, distro), repo_name, 'devel job data for repo')
 
-def load_repo_haros_data(repo_name, distro=None):
+def get_repo_reports_data(repo_name, distro=None):
     """
-    Load haros_data.yaml properties into dictionary for repo
-    @return: manifest properties dictionary
+    Search the rosdoc test restults / report directory for available reports
+    @return List of found reports, each report a dict containing "name" and "urls"
     @raise UtilException: if unable to load. Text of error message is human-readable
     """
-    return _load_manifest_file(repo_haros_data_file(repo_name, distro), repo_name, 'HAROS data for repo')
+    reportsdir = repo_report_dir(repo_name, distro)
+    if not os.path.exists(reportsdir):
+        return []
+    #
+    reportsdir_contents = os.listdir(reportsdir)
+    reports = []
+    # build the base URL for links
+    if distro:
+        reportsurl = doc_url + distro + '/devel_jobs/' + repo_name + "/test_results/"
+    else:
+        reportsurl = doc_url + 'devel_jobs/' + repo_name + "/test_results/"
+    #
+    for report_dir in reportsdir_contents:
+        report_path = os.path.join(reportsdir,report_dir)
+        if not os.path.isdir(report_path):
+            continue
+        # Beautify the name of known reports if possible
+        report_name = known_reports_name_mappings.get(report_dir, report_dir) 
+        # Find index file or report files (XML etc)
+        # in the reports folder and construct URLs to use in links.
+        report_contents = os.listdir(report_path)
+        # only consider files, not folders
+        report_contents = [e for e in report_contents if os.path.isfile(os.path.join(report_path, e))]
+        if len(report_contents)==0:
+            continue
+        # If there is an 'index' file, make a link to that,
```

Comment:
If [HAROS](https://github.com/git-afsantos/haros) is used, it will create an html test result.
Not sure if there are others who would also generate html test results instead of xml.